__tests__/lifecycle-scripts.js

```diff
@@ -142,3 +142,23 @@ test('should inherit existing environment variables when setting via yarnrc', as
   expect(stdout).toMatch(/^RAB$/m);
   expect(stdout).toMatch(/^FOO$/m);
 });
+
+test('should not show any error messages when script ends successfully', async () => {
+  const stdout = await execCommand('test', 'script-success');
+
+  expect(stdout).toMatch(/Done in /);
+});
+
+test('should show correct error message when the script ends with an exit code', async () => {
+  await execCommand('test', 'script-fail').catch(error => {
```

Comment:
This test would pass if no error is thrown (if the script doesn't fail). I'd use `expect().toThrow()` or even better, [toThrowErrorMatchingSnapshot()](https://facebook.github.io/jest/docs/expect.html#tothrowerrormatchingsnapshot)